### ISSUE TITLE
[Bug] 多卡 Plan 中 MLP-Medium MODEL 任务大量超时/卡死

### ISSUE THREAD

**Issue by chenxibj** [chenxibj/ai-hardware-verification-platform#512]
## 问题描述
4卡 Plan 中的 MLP-Medium 模型推理任务频繁超时或卡死。

## 数据
T4 (4卡MODEL-DP, Plan 2197):
- MLP-Medium/batch=1: ❌ never started
- MLP-Medium/batch=4: ❌ never started
- MLP-Medium/batch=8: ❌ never started
- MLP-Medium/batch=16: ❌ never started

T3 (4卡OP, Plan 2199):
- MLP-Medium/batch=16: ❌ 脚本执行失败 - no output for 120s (stuck)
- MLP-Medium/batch=4,8: ❌ never started

## 分析
- 13个 OPERATOR 任务全部正常通过
- 4个 MODEL (MLP-Medium) 任务全部失败
- 多卡 MODEL 推理在 DataParallel 模式下可能有兼容性问题
- 部分是调度问题(never started)，部分是执行问题(stuck 120s)

**Comment by chenxibj** [comment]
## 根因分析 & 修复方案

### 问题现象
4卡/2卡 Plan 中的 MLP-Medium MODEL 推理任务频繁卡死/超时（600s timeout killed）。
OPERATOR 任务正常，仅 MODEL 类型受影响。

### 根因
`model_inference.py` 的 `setup_model_for_inference()` 函数在多 GPU 场景下，对 `nn.Module` 实例（如 TorchMLP）使用 `torch.nn.DataParallel` 进行包装。

**DataParallel 在小模型上会导致进程死锁。** 具体表现：
- 第一次 forward pass 时，DataParallel 的 scatter/gather 机制在 GPU 间通信时无限阻塞
- 进程卡在 warmup 阶段，无任何输出，最终被超时机制 kill

GPU 机器上的日志证据：
```
[MULTI-GPU] DataParallel inference on 2 GPUs
# ← 之后没有任何输出，直到 600s 后被 kill
Process killed: timeout exceeded (600s)
```

手动在 L40S 机器上复现确认：DataParallel + MLP-Medium 的第一次 forward pass 必定死锁（30s 内无响应）。

### 修复方案（commit 8cfae7ca）

**移除 nn.Module 实例的 DataParallel 包装**：
- 对内置 benchmark 模型（TorchMLP 等），多 GPU 场景直接使用 `cuda:0` 单卡执行
- HuggingFace 字符串模型名仍然使用 `device_map="auto"` 进行多卡分布（不受影响）
- 返回 `gpu_count=1`，避免 batch size 被错误放大

### 验证
- ✅ 23 个相关单元测试全部通过
- ✅ GPU 机器实测：MLP-Medium + 2 GPUs 从死锁 → 2s 内完成，所有 batch size PASS
- ✅ 修改文件：`eval-scripts/model_inference.py`、3 个测试文件

**Comment by chenxibj** [comment]
Fixed in commit 8cfae7ca. DataParallel removed for nn.Module instances, verified on L40S GPU machine.
